passes/sat/sim.cc

```diff
@@ -2541,6 +2541,9 @@ struct AnnotateActivity : public OutputWriter {
 		double clk_period = real_timescale * (double)max_time / (clktoggleCounts[0] / 2.0);
 		double frequency = 1.0 / clk_period;
 		worker->top->module->set_string_attribute("$FREQUENCY", std::to_string(frequency));
+		worker->top->module->set_string_attribute("$DURATION", std::to_string(max_time));
+		// Timescale is scaled by 1e12 so it can be retrieved (not 0) in Python due to precision limitation during conversion
+		worker->top->module->set_string_attribute("$TIMESCALE", std::to_string(real_timescale * 1e12));
```

Comment:
Maybe the `to_string` method is the wrong one? Can you use a string formatting function where digits of precision is specifiable? Or something you can use scientific notation with?

Reply:
Correct, that was the wrong function to use. I opted for 
```
std::stringstream ss;
ss << std::setprecision(4) << real_timescale;

```